disdrodb/L0/run_disdrodb_l0_reader.py

```diff
@@ -0,0 +1,109 @@
+##################################################
+## Wrapper to run readers by command lines
+##################################################
+## Author: Régis Longchamp
+##################################################
+import click
+
+from disdrodb.L0 import click_L0_readers_options
+from disdrodb.L0.retrive_reader import get_reader
+
+
+# -------------------------------------------------------------------------.
+# Click Command Line Interface decorator
+@click.command()
+@click.argument("data_source", metavar="<data_source>")
+@click.argument("reader_name", metavar="<reader_name>")
+@click_L0_readers_options  # get default arguments
+def run_reader(
+    data_source: str,
+    reader_name: str,
+    raw_dir: str,
+    processed_dir: str,
+    l0a_processing: bool = True,
+    l0b_processing: bool = True,
+    keep_l0a: bool = False,
+    force: bool = False,
+    verbose: bool = False,
+    debugging_mode: bool = False,
+    lazy: bool = True,
+    single_netcdf: bool = True,
+) -> None:
+    """Wrapper to run reader functions by command lines.
+
+    Parameters
+    ----------
+    data_source : str
+        Institution name (when campaign data spans more than 1 country) or country (when all campaigns (or sensor networks) are inside a given country)
+    reader_name : str
+        Campaign name
+    raw_dir : str
+        Directory path of raw file for a specific campaign.
+        The path should end with <campaign_name>.
+        Example raw_dir: '<...>/disdrodb/data/raw/<campaign_name>'.
+        The directory must have the following structure:
+        - /data/<station_id>/<raw_files>
+        - /metadata/<station_id>.json
+        Important points:
+        - For each <station_id> there must be a corresponding JSON file in the metadata subfolder.
+        - The <campaign_name> must semantically match between:
+           - the raw_dir and processed_dir directory paths;
+           - with the key 'campaign_name' within the metadata YAML files.
+        - The campaign_name are set to be UPPER CASE.
+    processed_dir : str

```

Comment:
We should really revise the docstring of the reader. Most of them are out-of-date.
L0A_processing, L0B_processing and keep_L0A should be lowercase.
And the documentation of raw_dir and processed_dir is wrong !

Reply:
@ghiggi 
All reader() functions are different, but have the same argument by definition. Why not trying something like that : 

``` python 

def is_documented_by(original):
  def wrapper(target):
    target.__doc__ = original.__doc__
    return target
  return wrapper

class Waldo:
  @is_documented_by(Foo.getBar)
  def getBar():
    ...

```


But not sure it works with sphinx, should be tested.

Reply:
Agree

Reply:
@ghiggi I've added a generic docstring. Works with sphinx and help(reader) as well !

Reply:
check `is_documented_by` and `reader_generic_docstring` from `disdrodb\L0\L0_processing.py`